packages/plugins/i18n/server/src/services/content-types.ts

```diff
@@ -149,9 +154,17 @@ const getNestedPopulateOfNonLocalizedAttributes = (modelUID: any) => {
   const schema = strapi.getModel(modelUID);
   const scalarAttributes = getScalarAttributes(schema);
   const nonLocalizedAttributes = getNonLocalizedAttributes(schema);
-  const currentAttributesToPopulate = difference(nonLocalizedAttributes, scalarAttributes);
-  const attributesToPopulate = [...currentAttributesToPopulate];
 
+  const allAttributes = [...scalarAttributes, ...nonLocalizedAttributes];
+  if (schema.modelType === 'component') {
+    allAttributes.push(...getRelationalAttributes(schema));
+  }
+
+  const currentAttributesToPopulate = allAttributes.filter((value, index, self) => {
+    return self.indexOf(value) === index && self.lastIndexOf(value) === index;
+  });
```

Comment:
shouldn't compos be inside the non localized attributes list?

Reply:
The compos are yes (e.g. 'nonLocalizedCompo')

but I found that when this is called recursively to get the nested populate of a non localized component we need to explicitly populate that components relations (e.g. 'nonLocalizedCompo.tag'), otherwise they're missed by the top level populate

Reply:
makes sense 🫡